modules/azure-landing-zone/frontdoor.tf

```diff
@@ -211,3 +191,39 @@ resource "azurerm_frontdoor" "main" {
 
   depends_on = [azurerm_frontdoor_firewall_policy.custom, azurerm_key_vault_access_policy.frontdoor_kv_access]
 }
+
+resource "azurerm_frontdoor_custom_https_configuration" "https" {
+  for_each = { for frontend in var.frontends :
+    frontend.name => frontend
+    if lookup(frontend, "enable_ssl", true)
+  }
+
+  resource_group_name               = azurerm_frontdoor.main.resource_group_name
+  frontend_endpoint_id              = "/subscriptions/${var.subscription_id}/resourcegroups/${azurerm_frontdoor.main.resource_group_name}/providers/Microsoft.Network/frontdoors/${azurerm_frontdoor.main.name}/frontendendpoints/${each.value["name"]}"
+  custom_https_provisioning_enabled = true
+
+  custom_https_configuration {
+    certificate_source                         = var.ssl_mode
+    azure_key_vault_certificate_secret_name    = var.ssl_mode == "AzureKeyVault" ? data.azurerm_key_vault_secret.certificate[each.value["name"]].name : null
+    azure_key_vault_certificate_secret_version = var.ssl_mode == "AzureKeyVault" ? data.azurerm_key_vault_secret.certificate[each.value["name"]].version : null
+    azure_key_vault_certificate_vault_id       = var.ssl_mode == "AzureKeyVault" ? data.azurerm_key_vault.certificate_vault.id : null
+  }
+}
+
+resource "azurerm_frontdoor_custom_https_configuration" "https_www_redirect" {
+  for_each = { for frontend in var.frontends :
+    frontend.name => frontend
+    if lookup(frontend, "www_redirect", false)
```

Comment:
enable_ssl field should be enough to iterate through the frontends and enable/disable https based on a single flag. Why is www_redirect flag used again to to create the resource.

Reply:
ignore just saw that we do www redirect as well.